### PR TITLE
Sonar Analyses fixes

### PR DESCRIPTION
**Pull request by JuditKnoll**
I looked at the Sonar Cloud Analyses results of the [issues introduced this year](https://sonarcloud.io/project/issues?createdAfter=2023-01-01&resolved=false&id=com.github.spotbugs.spotbugs). This PR solves some of them.
I haven't added any changelog entry, since I don't think this type of fix needs it.

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [ ] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: eclipsePlugin/src/de/tobject/findbugs/builder/PDEClassPathGenerator.java [hunk 1]
```diff
@@ -90,18 +90,16 @@ private static Set<String> createJavaClasspath(IJavaProject javaProject) {
...
+                    if (classpathContainer instanceof JREContainer) {
+                        IClasspathEntry[] classpathEntries = classpathContainer.getClasspathEntries();
+                        for (IClasspathEntry iClasspathEntry : classpathEntries) {
+                            IPath path = iClasspathEntry.getPath();
+                            // smallest possible fix for #1228 Eclipse plugin always uses host VM to resolve JDK classes
+                            if (isValidPath(path) &&
+                                    ("rt.jar".equals(path.lastSegment())
+                                            || "jrt-fs.jar".equals(path.lastSegment())
+                                            || "jce.jar".equals(path.lastSegment()))) {
+                                classPath.add(path.toOSString());
```
Comment (JuditKnoll) [inline eclipsePlugin/src/de/tobject/findbugs/builder/PDEClassPathGenerator.java:102]: This is the only modification, where the fix is not the exact answer to the Sonar Analyses. It comes from here: [Merge this if statement with the enclosing one.](https://sonarcloud.io/project/issues?createdAfter=2023-01-01&resolved=false&id=com.github.spotbugs.spotbugs&open=AYuNJIc12YKIFvInB8dl), but the instanceof check makes the null check redundant.

**Review by ThrawnCA — APPROVED** [review]

**Review by hazendaz — APPROVED** [review]

**Comment by hazendaz** [comment]
Think we may benefit from adding open rewrite to our project.  See https://plugins.gradle.org/plugin/org.openrewrite.rewrite

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/ba/NullnessAnnotationTest.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
  *
  * @author kzaikin
  */
-public class NullnessAnnotationTest {
```
Comment (Pankraz76) [inline spotbugs-tests/src/test/java/edu/umd/cs/findbugs/ba/NullnessAnnotationTest.java:18]: - https://docs.openrewrite.org/recipes/java/recipes/rewritetestclassesshouldnotbepublic

File: spotbugs/src/main/java/edu/umd/cs/findbugs/detect/AbstractAssertDetector.java [hunk 3]
```diff
@@ -33,14 +33,14 @@ public abstract class AbstractAssertDetector extends OpcodeStackDetector {
...
 
-    public AbstractAssertDetector(BugReporter bugReporter) {
+    protected AbstractAssertDetector(BugReporter bugReporter) {
         this.bugReporter = bugReporter;
     }
 
     /**
      * Implement this method in a concrete detector
      */
-    abstract protected void detect(int seen);
```
Comment (Pankraz76) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/detect/AbstractAssertDetector.java:43]: - https://docs.openrewrite.org/recipes/staticanalysis/modifierorder


still incomplete:

Image

**Pull request merged**